test/canny.cpp

```diff
@@ -95,31 +96,31 @@ void cannyImageOtsuTest(string pTestFile, bool isColor)
 
     for (size_t testId=0; testId<testCount; ++testId) {
 
-        af_array inArray  = 0;
-        af_array outArray = 0;
-        af_array goldArray= 0;
-        dim_t nElems   = 0;
-
         inFiles[testId].insert(0,string(TEST_DIR "/CannyEdgeDetector/"));
         outFiles[testId].insert(0,string(TEST_DIR "/CannyEdgeDetector/"));
 
-        ASSERT_EQ(AF_SUCCESS, af_load_image(&inArray, inFiles[testId].c_str(), isColor));
-        ASSERT_EQ(AF_SUCCESS, af_load_image_native(&goldArray, outFiles[testId].c_str()));
-        ASSERT_EQ(AF_SUCCESS, af_get_elements(&nElems, goldArray));
+        try {
+            af_dtype type = (af_dtype)af::dtype_traits<T>::af_type;
+
+            array input = loadImage(inFiles[testId].c_str(), isColor).as(type);
+            array gold  = loadImage(outFiles[testId].c_str(), isColor).as(type);
+
+            dim_t nElems = gold.elements();
 
-        ASSERT_EQ(AF_SUCCESS, af_canny(&outArray, inArray, AF_CANNY_THRESHOLD_AUTO_OTSU, 0.08, 0.32, 3, false));
+            array output = canny(input, AF_CANNY_THRESHOLD_AUTO_OTSU, 0.08, 0.32, 3, false);
 
-        std::vector<char> outData(nElems);
-        ASSERT_EQ(AF_SUCCESS, af_get_data_ptr((void*)outData.data(), outArray));
+            output = (255.0*normalize<T>(output)).as(type);
 
-        std::vector<char> goldData(nElems);
-        ASSERT_EQ(AF_SUCCESS, af_get_data_ptr((void*)goldData.data(), goldArray));
+            std::vector<T> outData(nElems);
```

Comment:
T is char in the original. Again I don't know if this is the cause of the failures. Probably not.

Reply:
Failures are not related to this. I modified the test because it was not properly comparing values. I came across this issue while cross-referencing our output with ITK's.